test/r_tree/r_tree_test.dart

```diff
@@ -197,30 +213,105 @@ main() {
             tree.insert(item);
           }
         }
+        assertTreeHeightValidity(tree);
 
         var items = tree.search(Rectangle(0, 0, 50, 50));
         expect(items.length, equals(2500));
 
         data.forEach((RTreeDatum item) {
           tree.remove(item);
         });
+        assertTreeHeightValidity(tree);
 
         items = tree.search(Rectangle(0, 0, 50, 50));
         expect(items.length, equals(0));
 
         //test inserting after removal to ensure new root leaf node functions correctly
         tree.insert(RTreeDatum<String>(Rectangle(0, 0, 1, 1), 'New Initial Item'));
+        assertTreeHeightValidity(tree);
 
         items = tree.search(Rectangle(0, 0, 50, 50));
 
         items.forEach((datum) {
           expect(datum.value, equals('New Initial Item'));
         });
       });
+
+      test('remove all items and then reload', () {
+        final tree = RTree(3);
+
+        final items = <RTreeDatum<String>>[];
+        for (var i = 0; i < 20; i++) {
+          final item = RTreeDatum(Rectangle(0, i, 1, 1), 'Item $i');
+          items.add(item);
+          tree.insert(item);
+        }
+
+        for (final item in items) {
+          tree.remove(item);
+        }
+
+        tree.load(items.sublist(0, 3));
+      });
```

Comment:
This test reproduces the type error that would occur prior to this PR, specifically the `load` call on L254.